### DISCUSSION TITLE
Implement `Object.set_block_signals()` and `Node.set_process..()` functions as properties

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#4533] [Engine Core]
There are such powerful methods as `Object.set_block_signals()` and `Node.set_process..()`
But it would be even more convenient if I could disable signals and process from editor as usual properties, instead of doing that in a script.
And I could quickly and easily set these settings for individual instances, doing this in a script would require writing some if statements.
So I suggest implementing these methods as properties

**Comment by Calinou** [comment]
I don't know if that's a good idea. I'm worried that some users will uncheck the process checkbox, not realizing what it does, then complain online that their `_process()` function doesn't run.

The same concern applies to `Object.set_block_signals()`.

**Reply by me2beats** [reply]
>  I'm worried that some users will uncheck the process checkbox, not realizing what it does, then complain online that their _process() function doesn't run

I think this could be fixed in different ways
- implementing a separate properties category like "don't go there if you unsure!"
- adding some warning icon to the property and a confirmation dialog like "this setting will disable all signals, are you really sure?". And this confirmation dialog could be turned off in EditorSettings

**Comment by Mickeon** [comment]
I personally find Godot to be reasonably easy to access, and the "minimal" nature of some Nodes definitely is part of it. I'm glad and understand that not everything internal has to be a property because it's bound to bloat the Inspector with properties most users wouldn't know what to do with, 'cause their effect may not be immediate, may only be in effect during scripting, or may require much more intricate knowledge of the engine, like in this idea.

**Comment by aXu-AP** [comment]
In 4.0 you can set `process_mode` to `disabled` in editor which to my understanding achieves the same result as `set_process(false)`. Also this is visualized in scene tree by graying out the node.

As for blocking signals, if you have hooked the signals you likely want them to usually fire, only temporalily disabling them, which would need script anyway. For some cases (like disabling some trigger for testing) disabling the entire node via `process_mode` is sufficient.
